### DISCUSSION TITLE
Clarification on Databases

### DISCUSSION THREAD

**Discussion by sabamerchant** [nasa/fprime#1422] [General]
Does anyone have more information of whether a database for F’ exists? If not a database, a compiled list of command and telemetry data

I have read conflicting information of a database needing to be made, but also a few components that have similar functionalities. Would anyone be able to clarify if there is a document or a place in the code where this would be?

Thank you!

**Comment by danjwait** [comment]
@sabamerchant From what I understand, the commands and telemetry (and event messages) are defined on a per-component basis. So if you want to see what telemetry, commands, etc. are available, there should be something in the .fpp file for each component (like the [Math Component](https://github.com/nasa/fprime/blob/devel/docs/Tutorials/MathComponent/Tutorial.md#The-MathSender-Component)), then the handler is in the component .cpp file. The [FPP guide ](https://fprime-community.github.io/fpp/fpp-users-guide.html#Defining-Components_Commands)has more info on the .fpp version of this. 

Once you've build the application, there should be an .xml file in <app>/build-artifacts/<target>/dict with *everything* command, telemetry, and event in the application (all the components) ; that's the file that's read by fprime-gds.

Hope that helps!
